http-api/src/lib.rs

```diff
@@ -475,16 +505,43 @@ async fn history_handler(
             })
             .skip(page * per_page)
             .take(per_page)
-            .cloned()
+            .map(|header| {
+                let id = commit_hash_to_fingerprint(ctx.clone(), &header.sha1.to_string())
+                    .unwrap_or_default();
+                let peer = fingerprints
+                    .get(&id)
+                    .map(|(id, person)| -> Result<Peer, Error> {
+                        let info = project_info(project.to_owned(), ctx.paths.to_owned())?;
```

Comment:
Why are we fetching project info on every commit? It doesn't change.

Reply:
Yeah this came up because we needed the project info for checking delegates, and at some point I didn't want to create project info for every `history_handler` call.
But it's better than calling it for every commit that has peer information.
So I'll move it to the beginning of the `history_handler`